src/aws/flb_aws_util.c

```diff
@@ -296,8 +298,26 @@ struct flb_http_client *request_do(struct flb_aws_client *aws_client,
     }
 
     /* Add AWS Fluent Bit user agent */
-    ret = flb_http_add_header(c, "User-Agent", 10,
-                              "aws-fluent-bit-plugin", 21);
+    if (aws_client->extra_user_agent == NULL) {
+        ret = flb_http_add_header(c, "User-Agent", 10,
+                                  "aws-fluent-bit-plugin", 21);
+    } 
+    else {
+        user_agent_prefix = flb_sds_create_size(64);
+        tmp = flb_sds_printf(&user_agent_prefix, "aws-fluent-bit-plugin-%s", 
+                             aws_client->extra_user_agent);
+        if (!tmp) {
+            flb_errno();
+            flb_sds_destroy(user_agent_prefix);
+            flb_error("[aws_client] failed to fetch user agent");
+            goto error;
+        }
+        user_agent_prefix = tmp;
+
+        ret = flb_http_add_header(c, "User-Agent", 10, user_agent_prefix, 
```

Comment:
destroy` user_agent_prefix` after this line. `flb_http_add_header` will copy its memory to a new string, so you must free your copy otherwise its a memory leak